### DISCUSSION TITLE
TypeScript PrismaClient constructor type checking

### DISCUSSION THREAD

**Discussion by kennytraction** [prisma/prisma#14740] [Q&A]
Does type checking on `new PrismaClient()` take a while for anyone else? We have a fairly large scheme and build times take a while (2-3 minutes). Nearly all of that time is due to type checking when creating an instance of the PrismaClient. Has anyone found a way to reduce this time?

**Comment by nurul3101** [comment]
Hey @kennytraction 👋 
Do you mean it takes 2-3 minutes to generate your `PrismaClient`, when you execute `npx prisma generate`?
How many models does your schema have?

**Reply by kennytraction** [reply]
No this isn't generating the client. This is running `tsc` on the project. We have a single .ts file that initializes the `PrismaClient`.

Ex:
```
import { PrismaClient } from '@prisma/client'
export class MyPrisma {
  prisma: PrismaClient
  constructor() {
    this.prisma = new PrismaClient()  // This lines causes the 2-3 minutes to transpile
  }
}
```

We have 200+ models so the client is fairly large. I'm checking to see if anyone else has noticed this or if there is something else we should be doing to reduce the time it takes to build the project.

**Reply by pantharshit00** [reply]
Hello @kennytraction 

Sorry for late followup here. If you are still experiencing this, I would consider this a performance bug. You should report it here: https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
